src/amr/data/particles/refine/split_2d.h

```diff
@@ -0,0 +1,481 @@
+/*
+Splitting reference material can be found @
+  https://github.com/PHAREHUB/PHARE/wiki/SplitPattern
+*/
+
+#ifndef PHARE_SPLIT_2D_H
+#define PHARE_SPLIT_2D_H
+
+namespace PHARE::amr
+{
+using namespace PHARE::core;
+
+template<>
+struct ExactSplitParticleDispatcher<DimConst<2>>
+{
+    template<typename Deltas, typename DeltaVals>
+    static void set_deltas(Deltas& deltas, DeltaVals const& delta_vals, size_t const root)
+    {
+        for (size_t y = 0; y < root; y++)
+        {
+            size_t yoff = y * root;
+            for (size_t x = 0; x < root; x++)
+            {
+                deltas[x + yoff] = {delta_vals[x], delta_vals[y]};
+            }
+        }
+    }
+};
+
+
+/**************************************************************************/
+template<>
+struct SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<4>>
+{
+    template<typename Weights, typename WeightVal>
+    static void set_weights(Weights& weights, WeightVal const& weight_val)
+    {
+        for (uint8_t i = 0; i < 4; i++)
+            weights[i] = weight_val;
+    }
+
+    template<typename Deltas, typename DeltaVal>
+    static void set_deltas(Deltas& deltas, DeltaVal const& delta_val)
+    {
+        deltas[0] = {-delta_val, -delta_val};
+        deltas[1] = {-delta_val, +delta_val};
+        deltas[2] = {+delta_val, -delta_val};
+        deltas[3] = {+delta_val, +delta_val};
+    }
+};
+
+
+/**************************************************************************/
+template<>
+struct SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<8>, Shape::Cross>
+{
+    template<typename Weights, typename WeightVals>
+    static void set_weights(Weights& weights, WeightVals const& weight_vals)
+    {
+        for (uint8_t i = 0; i < 2; i++)
+            weights[i] = weight_vals[1];
+
+        for (uint8_t i = 2; i < 6; i++) // inner square
+            weights[i] = weight_vals[0];
+
+        for (uint8_t i = 6; i < 8; i++)
+            weights[i] = weight_vals[1];
+    }
+
+    template<typename Deltas, typename DeltaVals>
+    static void set_deltas(Deltas& deltas, DeltaVals const& delta_vals)
+    {
+        deltas[0] = {-delta_vals[1], -delta_vals[1]};
+        deltas[1] = {-delta_vals[1], +delta_vals[1]};
+
+        deltas[2] = {-delta_vals[0], -delta_vals[0]}; // inner square
+        deltas[3] = {-delta_vals[0], +delta_vals[0]};
+        deltas[4] = {+delta_vals[0], -delta_vals[0]};
+        deltas[5] = {+delta_vals[0], +delta_vals[0]};
+
+        deltas[6] = {+delta_vals[1], -delta_vals[1]};
+        deltas[7] = {+delta_vals[1], +delta_vals[1]};
+    }
```

Comment:
these 8 lines should be simplified : we use `delta_vals[0]` for a `PURPLE` pattern, and `delta_vals[1]` for a `PINK` one. As is, we have 2 `PURPLE` which is not correct, and not in agreement with the Wiki.

Reply:
removed/replaced by the single PINK/PURPLE dispatch